src/promptflow/promptflow/azure/_utils/gerneral.py

```diff
@@ -3,14 +3,56 @@
 # ---------------------------------------------------------
 
 import jwt
+from promptflow.exceptions import ValidationException
 
 
 def is_arm_id(obj) -> bool:
     return isinstance(obj, str) and obj.startswith("azureml://")
 
 
+def get_token(credential, resource):
+    from azure.ai.ml._azure_environments import _resource_to_scopes
+
+    azure_ml_scopes = _resource_to_scopes(resource)
+    token = credential.get_token(*azure_ml_scopes).token
+    # validate token has aml audience
+    decoded_token = jwt.decode(
+        token,
+        options={"verify_signature": False, "verify_aud": False},
+    )
+    if decoded_token.get("aud") != resource:
+        msg = """AAD token with aml scope could not be fetched using the credentials being used.
+            Please validate if token with {0} scope can be fetched using credentials provided to MLClient.
```

Comment:
PFClient?

Reply:
When will this happen?

Reply:
This is a reference to the V2 SDK, and I have not encountered this error so far.

Reply:
I will change MLClient to PFClient in the next PR to avoid the need for approval again.